Undefined property PhpParser\Node\Param::$flags

# Bug Report

<!-- First, thank you for reporting a bug. That takes time and we appreciate that! -->

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | ^0.7.41 or 0.8.x-dev@47fc574 (docker image)              |
| Installed as   | composer dependency, composer bin dependency, docker image                      |

```
Notice: Undefined property: PhpParser\Node\Param::$flags in /usr/src/app/api/vendor/nikic/php-parser/lib/PhpParser/PrettyPrinter/Standard.php on line 21
```
and
```
 [ERROR] Could not process "src/Services/Product/TaxService.php" file, due to:  
         "Argument 1 passed to PhpParser\PrettyPrinterAbstract::pModifiers()    
         must be of the type int, null given, called in                         
         /usr/src/app/api/vendor/nikic/php-parser/lib/PhpParser/PrettyPrinter/St
         andard.php on line 21".
```

My rector.yaml:
```yaml
services:
    Rector\Php74\Rector\Property\TypedPropertyRector:

parameters:
    auto_import_names: true
    import_short_classes: false
    autoload_paths:
        - 'vendor/autoload.php'
    paths:
        - 'src'
```

What is going on? Notice and error message happen tens of times on a single rector run.
The same happens when installing rector as vendor-bin dependency (using bamarni/composer-bin-plugin).
Third option I tried was docker. It works with --dry run (`docker run -it --rm -v $(pwd):/project rector/rector:latest process --config /project/rector.yaml --dry-run`, P.S. yes, config paths were updated to match /project) but without dry run an error occurs:
```
Fatal error: Declaration of Symfony\Component\Cache\Adapter\ArrayAdapter::clear() must be compatible with Symfony\Component\Cache\Adapter\AdapterInterface::clear(string $prefix = '') in /project/vendor/symfony/cache/Adapter/ArrayAdapter.php on line 76
``` 
My first hint to problem with ArrayAdapter fatal error is: My project uses SF ^4.4.*, but rector from latest image uses ^5.0.6. It may conflict when I add vendor/autoload.php to the rector.yaml but shouldn't rector handle that? If not then there should be a statement in docs not to use rector version that has dependencies not matching developer's project dependencies.

--- a/composer.json
+++ b/composer.json
@@ -18,7 +18,7 @@
         "jean85/pretty-package-versions": "^1.2",
         "nette/robot-loader": "^3.2",
         "nette/utils": "^3.1",
-        "nikic/php-parser": "^4.7",
+        "nikic/php-parser": "4.6",
         "ondram/ci-detector": "^3.4",
         "phpstan/phpdoc-parser": "^0.4.7",
         "phpstan/phpstan-phpunit": "^0.12.10",

--- a/composer.json
+++ b/composer.json
@@ -18,7 +18,7 @@
         "jean85/pretty-package-versions": "^1.2",
         "nette/robot-loader": "^3.2",
         "nette/utils": "^3.1",
-        "nikic/php-parser": "4.5",
+        "nikic/php-parser": "^4.7",
         "ondram/ci-detector": "^3.4",
         "phpstan/phpdoc-parser": "^0.4.7",
         "phpstan/phpstan-phpunit": "^0.12.10",
@@ -37,7 +37,7 @@
         "symplify/set-config-resolver": "^8.1.15",
         "symplify/smart-file-system": "^8.1.15",
         "tracy/tracy": "^2.7",
-        "phpstan/phpstan": "0.12.32"
+        "phpstan/phpstan": "^0.12.33"
     },
     "require-dev": {
         "friendsofphp/php-cs-fixer": "^2.16",

--- a/packages/node-type-resolver/src/PHPStan/Scope/PHPStanNodeScopeResolver.php
+++ b/packages/node-type-resolver/src/PHPStan/Scope/PHPStanNodeScopeResolver.php
@@ -4,6 +4,7 @@ declare(strict_types=1);
 
 namespace Rector\NodeTypeResolver\PHPStan\Scope;
 
+use Nette\Utils\Strings;
 use PhpParser\Node;
 use PhpParser\Node\Stmt\Class_;
 use PhpParser\Node\Stmt\ClassLike;
@@ -177,14 +178,20 @@ final class PHPStanNodeScopeResolver
     }
 
     /**
-     * @param Class_|Interface_ $classOrInterfaceNode
+     * @param Class_|Interface_ $classLike
      */
     private function resolveClassOrInterfaceScope(
-        Node $classOrInterfaceNode,
+        ClassLike $classLike,
         MutatingScope $mutatingScope
     ): MutatingScope {
-        $className = $this->resolveClassName($classOrInterfaceNode);
-        $classReflection = $this->reflectionProvider->getClass($className);
+        $className = $this->resolveClassName($classLike);
+
+        // is anonymous class? - not possible to enter it since PHPStan 0.12.33, see https://github.com/phpstan/phpstan-src/commit/e87fb0ec26f9c8552bbeef26a868b1e5d8185e91
+        if ($classLike instanceof Class_ && Strings::match($className, '#^AnonymousClass(\w+)#')) {
+            $classReflection = $this->reflectionProvider->getAnonymousClassReflection($classLike, $mutatingScope);
+        } else {
+            $classReflection = $this->reflectionProvider->getClass($className);
+        }
 
         return $mutatingScope->enterClass($classReflection);
     }

--- a/phpstan.neon
+++ b/phpstan.neon
@@ -386,3 +386,4 @@ parameters:
         - '#Method Rector\\Nette\\NodeResolver\\FormVariableInputNameTypeResolver\:\:resolveFromGetComponentMethodCall\(\) should return array<string, string\> but returns array<int\|string, string\>#'
         - '#Method Rector\\Core\\PhpParser\\Node\\BetterNodeFinder\:\:findPreviousAssignToExpr\(\) should return PhpParser\\Node\\Expr\\Assign\|null but returns PhpParser\\Node\|null#'
         - '#Method Rector\\Nette\\FormControlTypeResolver\\GetComponentMethodCallFormControlTypeResolver\:\:resolve\(\) should return array<string, string\> but returns array<int\|string, string\>#'
+        - '#Class with base "LexerFactory" name is already used in "PHPStan\\Parser\\LexerFactory", "Rector\\Core\\PhpParser\\Parser\\LexerFactory"\. Use unique name to make classes easy to recognize#'

--- a/rules/dead-code/src/Rector/StaticCall/RemoveParentCallWithoutParentRector.php
+++ b/rules/dead-code/src/Rector/StaticCall/RemoveParentCallWithoutParentRector.php
@@ -93,7 +93,7 @@ PHP
         }
 
         if ($this->isAnonymousClass($classLike)) {
-            //currently the classMethodManipulator isn't able to find usages of anonymous classes
+            // currently the classMethodManipulator isn't able to find usages of anonymous classes
             return null;
         }
 

